### PR TITLE
[Easy] OrderIds are u16

### PR DESCRIPTION
**Pull request by bh2smith**
An easy mistake (since `orderId` corresponds to an array index) is that `orderId` is `uint256`. However, for space saving reasons, we have decided to restrict this value to `uint16` (allows ~65K order per account.. which should suffice). 

**Test Plan:** Unit tests

### CODE REVIEW

File: contracts/BatchExchange.sol [hunk 1]
```diff
@@ -233,7 +233,7 @@ contract BatchExchange is EpochTokenLocker {
       * @param sellAmounts maximum amounts of sell token to be exchanged in new orders
       * @return an array of indices in which `msg.sender`'s new orders are included
       *
-      * Emits {OrderCancelation} events for all cancelled orders and {OrderPlacement} events with all relevant new order details.
+      * Emits {OrderCancellation} events for all cancelled orders and {OrderPlacement} events with relevant new order details.
```
Comment (nlordell) [inline contracts/BatchExchange.sol]: Personally, I always go for Canadian/British spelling, so I prefer "cancelled". If we want to use American spelling, then we should change all instances of "cancelled" to "canceled" as well. Note that this change should be in a separate PR if we decide to do the change.

**Review by nlordell — APPROVED** [review]

File: contracts/BatchExchange.sol (hunk 1 above)
Comment (bh2smith) [reply contracts/BatchExchange.sol]: I also prefer this in some situations. However I prefer to use a Z in words like realization and stuff like that.

Comment (bh2smith) [reply contracts/BatchExchange.sol]: Yes... I agree separate PR... I wish I had made more specific commits.

**Review by fleupold — APPROVED** [review]
🍨

**Comment by anxolin** [comment]
Cool, thanks for the fast response @bh2smith
